adapter/src/dbt/adapters/fal/teleport_support/snowflake.py

```diff
@@ -0,0 +1,66 @@
+from typing import List
+from dbt.adapters.fal.connections import TeleportCredentials
+from dbt.fal.adapters.teleport.impl import TeleportAdapter
+from dbt.fal.adapters.teleport.info import TeleportInfo
+from dbt.adapters.fal.adapter_support import new_connection
+from dbt.adapters.base.relation import BaseRelation
+from dbt.adapters.base.impl import BaseAdapter
+
+
+class SnowflakeAdapterTeleport(TeleportAdapter):
+
+    def __init__(self, db_adapter: BaseAdapter, teleport_credentials: TeleportCredentials):
+        self._db_adapter = db_adapter
+        self._credentials = teleport_credentials
+        url = f's3://{teleport_credentials.s3_bucket}/teleport'
+
+        from dbt.adapters.fal.adapter_support import new_connection
+
+        with new_connection(self._db_adapter, "fal-snowflake:setup-teleport") as conn:
+            cur = conn.handle.cursor()
+
+            create_stage_query = f"""CREATE OR REPLACE STAGE falstage
```

Comment:
Couldn't put this in a macro

Reply:
Super curious about why we couldnt. It would not render it? The data passed have errors?

Reply:
It would not render it and just get stuck. This is before any logs start printing